### PR TITLE
Rahul/unsigned transaction in signing package

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Creates a new signing package struct. 
Adds `unsignedTransaction` to the signing package. 
This new struct is passed between the rust/ typescript layer instead of the frost SigningPackage. 
This change has also led to simplification on the RPC and CLI side for creating signature shares. 
Removes the need to pass the transaction hash of the unsigned transaction

Next step is to remove `unsignedTransaction` from the `aggregateSigningShare` RPC

## Testing Plan

Updated unit tests

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-rust/src/frost_utils/signing_package.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        Ok(vec)
+    }
+
+    pub fn deserialize(bytes: &[u8]) -> Result<SigningPackage, IronfishError> {
+        let mut signing_package_length_bytes = [0u8; 4];
+        signing_package_length_bytes.copy_from_slice(&bytes[..4]);
+
+        let length = u32::from_le_bytes(signing_package_length_bytes);
+
+        let frost_signing_package_bytes = &bytes[4..(4 + length as usize)];
```
Comment (hughy) [inline ironfish-rust/src/frost_utils/signing_package.rs:40]: the byte offsets in here look a bit odd

i wonder if it might be cleaner to follow the reader/writer pattern that we've used in other serializations like `SigningCommitment`

Comment (patnir) [reply ironfish-rust/src/frost_utils/signing_package.rs:40]: This is more consistent with Signature Share. I don't think doing read/ write will help with the "oddness". I think the core issue is that we don't know how large the unsigned transaction is going to be

Comment (patnir) [reply ironfish-rust/src/frost_utils/signing_package.rs:40]: I'm not opposed to changing it to the read/ write pattern.

Comment (hughy) [reply ironfish-rust/src/frost_utils/signing_package.rs:40]: > I don't think doing read/ write will help with the "oddness".

I think using a reader will let us rely on a cursor for reading through the byte array rather than slicing the array with indices.  It might make it a bit easier to follow since that's the pattern that most of our encodings use

Comment (patnir) [reply ironfish-rust/src/frost_utils/signing_package.rs:40]: I'll give that a go

File: ironfish-rust/src/frost_utils/signing_package.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        // write length
+        vec.extend_from_slice(&length_bytes);
+        // write frost signing package
+        vec.extend_from_slice(&serialized_frost_signing_package);
+
+        let mut transaction_vec = vec![];
+        // write unsigned transaction
+        self.unsigned_transaction.write(&mut transaction_vec)?;
+
+        vec.extend_from_slice(&transaction_vec);
```
Comment (andiflabs) [inline ironfish-rust/src/frost_utils/signing_package.rs:29]: This could be written directly into the `vec` vector:

```rust
self.unsigned_ransaction.write(&mut vec);
```

File: ironfish-rust/src/frost_utils/signing_package.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    pub unsigned_transaction: UnsignedTransaction,
+    pub frost_signing_package: FrostSigningPackage,
+}
+
+impl SigningPackage {
+    pub fn serialize(&self) -> Result<Vec<u8>, IronfishError> {
+        let serialized_frost_signing_package = self.frost_signing_package.serialize()?;
+        let length_bytes = (serialized_frost_signing_package.len() as u32).to_le_bytes();
+
+        let mut vec: Vec<u8> = vec![];
```
Comment (andiflabs) [inline ironfish-rust/src/frost_utils/signing_package.rs:19]: Nit: `let mut vec = Vec::<u8>::new()` (or even `with_capacity`)

File: ironfish-rust/src/frost_utils/signing_package.rs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#[derive(Clone)]
+pub struct SigningPackage {
+    pub unsigned_transaction: UnsignedTransaction,
+    pub frost_signing_package: FrostSigningPackage,
+}
+
+impl SigningPackage {
+    pub fn serialize(&self) -> Result<Vec<u8>, IronfishError> {
+        let serialized_frost_signing_package = self.frost_signing_package.serialize()?;
+        let length_bytes = (serialized_frost_signing_package.len() as u32).to_le_bytes();
```
Comment (andiflabs) [inline ironfish-rust/src/frost_utils/signing_package.rs:17]: Nit: although unlikely, this cast can truncate the length. I'd use `TryFrom` or `TryInto` for safety:

```rust
u32::try_from(serialized_frost_signing_package.len())
```

**Review by hughy — APPROVED** [review]

File: ironfish/src/rpc/routes/wallet/createAccount.ts [hunk 5]
```diff
@@ -46,23 +45,24 @@ routes.register<typeof CreateAccountRequestSchema, CreateAccountResponse>(
...
-      }
-      throw e
+    const name = request.data.name
+
+    if (context.wallet.accountExists(name)) {
+      throw new RpcValidationError(
+        `There is already an account with the name ${name}`,
+        400,
+        RPC_ERROR_CODES.ACCOUNT_EXISTS,
+      )
```
Comment (andiflabs) [inline ironfish/src/rpc/routes/wallet/createAccount.ts:55]: Was this change intentional? This introduces a concurrency problem (two concurrent calls with the same `name` may both pass the `accountExists` test and then both call `createAccount`, resulting in one call overriding the other without giving any hint of what occurred to the user)

Comment (patnir) [reply ironfish/src/rpc/routes/wallet/createAccount.ts]: unintentional, reverting it. merge got messed up somehow
